src/sentry/static/sentry/app/views/settings/components/settingsLayout.tsx

```diff
@@ -36,55 +109,90 @@ function SettingsLayout(props: Props) {
         </SettingsHeader>
 
         <MaxWidthContainer>
-          {typeof renderNavigation === 'function' && (
-            <SidebarWrapper>{renderNavigation()}</SidebarWrapper>
+          {shouldRenderNavigation && (
+            <SidebarWrapper isVisible={navVisible} offsetTop={navOffsetTop}>
+              {renderNavigation!()}
```

Comment:
wouldn't it be better to check if this function is available otherwise make it required?